iceoryx_posh/include/iceoryx_posh/popo/sample.hpp

```diff
@@ -146,20 +148,33 @@ class Sample
     ///
     const T* get() const noexcept;
 
+    using ConditionalConstChunkHeader_t = typename cxx::add_const_conditionally<mepoo::ChunkHeader, T>::type;
     ///
     /// @brief header Retrieve the header of the underlying memory chunk loaned to the sample.
     /// @return The ChunkHeader of the underlying memory chunk.
-    /// @details Only available for non-const type T.
     ///
-    template <typename S = T, typename = ForPublisherOnly<S, T>>
-    mepoo::ChunkHeader* getHeader() noexcept;
+    ConditionalConstChunkHeader_t* getHeader() noexcept;
 
     ///
     /// @brief header Retrieve the header of the underlying memory chunk loaned to the sample.
     /// @return The const ChunkHeader of the underlying memory chunk.
     ///
     const mepoo::ChunkHeader* getHeader() const noexcept;
 
+    ///
+    /// @brief Retrieve the custom header of the underlying memory chunk loaned to the sample.
+    /// @return The custom header of the underlying memory chunk.
+    ///
+    template <typename R = H, typename = HasCustomHeader<R, H>>
+    R& getCustomHeader() noexcept;
```

Comment:
Do we also require a `using ConditionConstCustomHeader`?

Reply:
No, since there is the static assert on line 73 which ensures that `T` and `H` are equal in their const qualifiers.

The `ConditionalConstChunkHeader_t` is necessary since the sample can be constructed from a publisher with non-const `T` or from a subscriber with `const T`. When it's constructed from the publisher it mutable, at least that's the current state. There is a task item to restrict the access to the `ChunkHeader`.

I'll add this as documentation to the `ConditionalConstChunkHeader_t`